src/main/java/uk/ac/ebi/atlas/download/FileDownloadController.java

```diff
@@ -144,4 +153,58 @@ public FileDownloadController(ExperimentFileLocationService experimentFileLocati
             zipOutputStream.close();
         }
     }
+
+    @RequestMapping(value = "json/experiments/check/zip",
+            method = RequestMethod.GET,
+            produces = "application/json;charset=UTF-8")
+    @ResponseStatus(HttpStatus.OK)
+    @ResponseBody
+    public String
+    checkMultipleExperimentsFileValid(@RequestParam(value = "accession", defaultValue = "") List<String> accessions)
+    {
+        List<Experiment> experiments = new ArrayList<>();
+        for (var accession : accessions) {
+            try {
+                experiments.add(experimentTrader.getPublicExperiment(accession));
+            } catch (Exception e) {
+                LOGGER.debug("Invalid experiment accession: {}", accession);
+            }
+        }
+        List<ExperimentFileType> fileTypeCheckList = new ArrayList<>(Arrays.asList(
+                ExperimentFileType.QUANTIFICATION_RAW,
+                ExperimentFileType.NORMALISED,
+                ExperimentFileType.EXPERIMENT_DESIGN));
+
+       Map<String, List<String>> inValidFilesList = new HashMap<>();
+
+        if (!experiments.isEmpty()) {
+            for (var experiment : experiments) {
+                List<String> invalidFiles = new ArrayList<>();
+                for (var fileType : fileTypeCheckList){
+
+                    if (fileType == ExperimentFileType.EXPERIMENT_DESIGN &&
+                            !experimentFileLocationService.getFilePath(experiment.getAccession(), fileType).toFile().exists()) {
+                        invalidFiles.add(fileType.getId());
+                    }
+
+                    else if (fileType != ExperimentFileType.EXPERIMENT_DESIGN) {
+                        var paths = experimentFileLocationService.getFilePathsForArchive(experiment.getAccession(), fileType);
+                        for (var path : paths){
+                            if(path.toFile().exists() == false) {
+                                invalidFiles.add(path.getFileName().toString());
+                            }
+                        }
+                    }
+                }
```

Comment:
Use streams with `filter` and `map`/`flatmap` to do all these in a more straightforward way.

Reply:
I amend the code with `filter` and `map`.